src/device.rs

```diff
@@ -6,6 +6,45 @@ use crate::driver::{clint, ClintDriver};
 
 // ———————————————————————————— Virtual Devices ————————————————————————————— //
 
+// Represents different data widths:
+//  - `Byte`: 8 bits (1 byte)
+//  - `Byte2`: 16 bits (2 bytes)
+//  - `Byte4`: 32 bits (4 bytes)
+//  - `Byte8`: 64 bits (8 bytes)
```

Comment:
Small nit: in Rust there are two types of comments, standard comments with `//`, and documentation comments with `///`.
The difference is that documentation comments (doc comments for short) are used to generate documentation automatically. You can try it on you machine, simply run `cargo doc --open` in the Miralis folder and navigate a bit around :)

The general rule is: when you add comments for documentation purposes (e.g. on top of functions or structures) use doc comments (`///`), that way the documentation can be auto-generated.